### ISSUE TITLE
Solution found for training on tf.keras 2.4

### ISSUE THREAD

**Issue by vangl34** [matterport/Mask_RCNN#2656]
To get mrcnn working, we modified the model.py file.
At the top of file, add the following class (you'll have to fix the layout).
`class AnchorsLayer(KL.Layer):
         def __init__(self, anchors, name="anchors", **kwargs):
        super(AnchorsLayer, self).__init__(name=name, **kwargs)
        self.anchors = tf.Variable(anchors)
   def call(self, dummy):
        return self.anchors
    def get_config(self):
        config = super(AnchorsLayer, self).get_config()
        return config
`

Next, instead of having `self.keras_model._losses = []`, replace it with `self.keras_model._losses.clear()`.

Lastly, instead of having `self.keras_model.add_metric(loss, name)`, replace it with `self.keras_model.add_metric(value=loss, aggregation='mean', name=name)`

These modifications were required for mrcnn to work for us. We hope these changes help fix your issues as well!

**Comment by jamalihuzaifa9** [comment]
@vangl34 Can you tell me about your system configuration such as OS version, GPU model, CUDA, and CUDNN version. I am struggling with the implementation on tf 2.x, the algorithm gets stuck at 1st epoch without any output or errors.

**Comment by vangl34** [comment]
> @vangl34 Can you tell me about your system configuration such as OS version, GPU model, CUDA, and CUDNN version. I am struggling with the implementation on tf 2.x, the algorithm gets stuck at 1st epoch without any output or errors.

Hmm, if you try ctrl + c to force interrupt, does it appear to be in a deadlock? Or does it say something else?

Linux , GeForce RTX 2080, CUDA 11.0, cudnn 8.0.4.30

**Comment by van-flow** [comment]
> @vangl34 Can you tell me about your system configuration such as OS version, GPU model, CUDA, and CUDNN version. I am struggling with the implementation on tf 2.x, the algorithm gets stuck at 1st epoch without any output or errors.

I am running into the same problem. Have you found a solution?
